src/main/java/li/cil/scannable/client/ScannableClient.java

```diff
@@ -0,0 +1,31 @@
+package li.cil.scannable.client;
+
+import dev.architectury.registry.menu.MenuRegistry;
+import li.cil.scannable.client.gui.ConfigurableBlockScannerModuleContainerScreen;
+import li.cil.scannable.client.gui.ConfigurableEntityScannerModuleContainerScreen;
+import li.cil.scannable.client.gui.ScannerContainerScreen;
+import li.cil.scannable.client.renderer.OverlayRenderer;
+import li.cil.scannable.client.scanning.ScanResultProviders;
+import li.cil.scannable.client.shader.Shaders;
+import li.cil.scannable.common.container.Containers;
+import net.fabricmc.api.ClientModInitializer;
+import net.fabricmc.fabric.api.client.event.lifecycle.v1.ClientTickEvents;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderEvents;
+import net.minecraft.client.gui.screens.MenuScreens;
+
+public final class ScannableClient implements ClientModInitializer {
+    @Override
+    public void onInitializeClient() {
+        ScanResultProviders.initialize();
+
+        MenuRegistry.registerScreenFactory(Containers.SCANNER_CONTAINER.get(), ScannerContainerScreen::new);
+        MenuRegistry.registerScreenFactory(Containers.BLOCK_MODULE_CONTAINER.get(), ConfigurableBlockScannerModuleContainerScreen::new);
+        MenuRegistry.registerScreenFactory(Containers.ENTITY_MODULE_CONTAINER.get(), ConfigurableEntityScannerModuleContainerScreen::new);
+
+        //WorldRenderEvents.BEFORE_DEBUG_RENDER.register(OverlayRenderer::onOverlayRender);
```

Comment:
WIP or can this be removed?

Reply:
Yes, this can be removed. I ended up using a mixin for that renderer because Fabric didn't have a good hook.